### PR TITLE
Provide an option to clear the consumer view

### LINKED ISSUE #40: Provide an option to clear the consumer view
When a consumer has logged hundreds of messages, it's a bit difficult to navigate through all. Having an option to clear the view (like the console view has) would be useful IMHO.

### PR DESCRIPTION
**Pull request by angelozerr**
Provide an option to clear the consumer view

Fixes #40

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ All notable changes to Kafka extension will be documented in this file.
 - Selected Cluster or Topic can now be deleted via the Delete shortcut (Cmd+Backspace on Mac). See [#79](https://github.com/jlandersen/vscode-kafka/issues/79)
 - Added SASL/SCRAM-256 and SASL/SCRAM-512 authentication support. See [#3](https://github.com/jlandersen/vscode-kafka/issues/3).
 - Added the option to enable basic SSL support for clusters without authentication. See [#84](https://github.com/jlandersen/vscode-kafka/issues/84).
+- The consumer view provides now a clear consumer command. See [#84](https://github.com/jlandersen/vscode-kafka/issues/40).
```
Comment (fbricon) [inline CHANGELOG.md:12]: The consumer view now provides a `Clear Consumer View` command

File: package.json [hunk 2]
```diff
@@ -270,6 +271,15 @@
                     "light": "images/light/toggle.svg",
                     "dark": "images/dark/toggle.svg"
                 }
+            },
+            {
+                "command": "vscode-kafka.consumer.clear",
+                "title": "Clear Consumer",
```
Comment (fbricon) [inline package.json:277]: Clear Consumer View

File: src/commands/consumers.ts [hunk 3]
```diff
@@ -75,6 +79,25 @@ export class ToggleConsumerCommandHandler {
         }
     }
 }
+export class ClearConsumerCommandHandler {
```
Comment (fbricon) [inline src/commands/consumers.ts:82]: ClearConsumerViewCommandHandler

**Review by fbricon — CHANGES_REQUESTED** [review]

**Comment by angelozerr** [comment]
This PR adds `Clear Consumer` command, see the following demo:

ClearConsumerDemo2

File: CHANGELOG.md (hunk 1 above)
Comment (angelozerr) [reply CHANGELOG.md]: fixed

File: package.json (hunk 2 above)
Comment (angelozerr) [reply package.json]: fixed

File: src/commands/consumers.ts (hunk 3 above)
Comment (angelozerr) [reply src/commands/consumers.ts]: fixed

**Comment by angelozerr** [comment]
I have updated the documentation.

File: docs/Consuming.md [hunk 4]
```diff
@@ -1,10 +1,44 @@
 # Consuming messages
 
-Consuming topics can be done by right clicking a topic in the explorer or from the command palette.  Some things to note about consuming:
+Consuming topics can be done:
+
+* with [Kafka Explorer](#kafka-explorer) by right clicking a topic in the explorer.
```
Comment (fbricon) [inline docs/Consuming.md:5]: `from the [Kafka Explorer](#kafka-explorer), by right-clicking on a topic and selecting "Start Consumer".`

File: docs/Consuming.md [hunk 5]
```diff
@@ -1,10 +1,44 @@
 # Consuming messages
 
-Consuming topics can be done by right clicking a topic in the explorer or from the command palette.  Some things to note about consuming:
+Consuming topics can be done:
+
+* with [Kafka Explorer](#kafka-explorer) by right clicking a topic in the explorer.
+* with the [Start command](#start-command) from the command palette.
```
Comment (fbricon) [inline docs/Consuming.md:6]: `from the [Start command](#start-command), from the command palette.`

File: docs/Consuming.md [hunk 6]
```diff
@@ -1,10 +1,44 @@
...
+Consuming topics can be done:
+
+* with [Kafka Explorer](#kafka-explorer) by right clicking a topic in the explorer.
+* with the [Start command](#start-command) from the command palette.
+
+## Consume with ...
+
+### Kafka explorer
+
+You can start consumer with [Kafka Explorer](Explorer.md#explorer) by right clicking a topic in the explorer:
```
Comment (fbricon) [inline docs/Consuming.md:12]: `You can start consuming messages from the [Kafka Explorer](Explorer.md#explorer), by right-clicking on a topic:`

File: docs/Consuming.md [hunk 7]
```diff
@@ -1,10 +1,44 @@
...
+
+![Start Consumer with Explorer](assets/start-consumer-from-explorer.png)
+
+Once this command is executed, it creates a consumer group (with generated id) and opens the [Consumer View](#consumer-view) where you can see the producing messages:
```
Comment (fbricon) [inline docs/Consuming.md:16]: `Once this command is launched, it creates a consumer group (with an auto-generated id), and opens the [Consumer View](#consumer-view) where you can see the messages being consumed:`

File: docs/Consuming.md [hunk 8]
```diff
@@ -1,10 +1,44 @@
...
+
+![Consumer group / Consumer View](assets/consumer-group-after-starting-from-explorer.png)
+
+In this case, you cannot configure offset, partition and consumer group id. The offset can be just configured with the [kafka.consumers.offset](#kafkaconsumersoffset) preference.
```
Comment (fbricon) [inline docs/Consuming.md:20]: `In this case, the starting offset can be only be configured via the [kafka.consumers.offset](#kafkaconsumersoffset) preference.`

File: docs/Consuming.md [hunk 9]
```diff
@@ -1,10 +1,44 @@
...
+
+Some things to note about consuming:
```
Comment (fbricon) [inline docs/Consuming.md:22]: Known limitations:

File: docs/Consuming.md [hunk 10]
```diff
@@ -1,10 +1,44 @@
...
 * UTF-8 encoded keys and values only. If data is encoded differently, it will not be pretty.
 * One consumer group is created per topic (may change in the future to just have one for the extension).
 
+### Start command
+
+![Start Consumer with palette](assets/start-consumer-from-command.png)
+
+## Consumer View
+
+`Consumer View` is a read-only editor which shows producing messages for a given topic:
```
Comment (fbricon) [inline docs/Consuming.md:33]: The `Consumer View` is a read-only editor which shows consumed messages for a given topic:

File: docs/Consuming.md [hunk 11]
```diff
@@ -1,10 +1,44 @@
...
+
+![Consumer view](assets/consumer-view.png)
+
+This editor provides 2 commands on the right of the editor:
```
Comment (fbricon) [inline docs/Consuming.md:37]: ... on the top right of the editor:

File: docs/Consuming.md [hunk 12]
```diff
@@ -1,10 +1,44 @@
...
+
+ * `Clear Consumer View` to clear the content of the view.
```
Comment (fbricon) [inline docs/Consuming.md:39]: `Clear Consumer View`: clears the view.

File: docs/Consuming.md [hunk 13]
```diff
@@ -1,10 +1,44 @@
...
+ * `Start/Stop` to stop or restart the consumer.
```
Comment (fbricon) [inline docs/Consuming.md:40]: `Start/Stop`: to stop or (re)start the consumer.

**Review by fbricon — CHANGES_REQUESTED** [review]

File: docs/Consuming.md [hunk 14]
```diff
@@ -1,10 +1,44 @@
...
+![Consumer group / Consumer View](assets/consumer-group-after-starting-from-explorer.png)
+
+In this case, the starting offset can be only be configured via the [kafka.consumers.offset](#kafkaconsumersoffset) preference.
+
+Known limitations:
 
 * UTF-8 encoded keys and values only. If data is encoded differently, it will not be pretty.
 * One consumer group is created per topic (may change in the future to just have one for the extension).
 
+### Start command
```
Comment (fbricon) [inline docs/Consuming.md:27]: Start Consumer command

File: docs/Consuming.md [hunk 15]
```diff
@@ -1,10 +1,44 @@
 # Consuming messages
 
-Consuming topics can be done by right clicking a topic in the explorer or from the command palette.  Some things to note about consuming:
+Consuming topics can be done:
+
+* from the [Kafka Explorer](#kafka-explorer), by right-clicking on a topic and selecting "Start Consumer".
+* from the [Start command](#start-command), from the command palette.
```
Comment (fbricon) [inline docs/Consuming.md:6]: `[Start Consumer](#start-consumer-command)` command

File: docs/Consuming.md [hunk 16]
```diff
@@ -1,10 +1,44 @@
...
+In this case, the starting offset can be only be configured via the [kafka.consumers.offset](#kafkaconsumersoffset) preference.
+
+Known limitations:
 
 * UTF-8 encoded keys and values only. If data is encoded differently, it will not be pretty.
 * One consumer group is created per topic (may change in the future to just have one for the extension).
 
+### Start command
+
+![Start Consumer with palette](assets/start-consumer-from-command.png)
```
Comment (fbricon) [inline docs/Consuming.md:29]: Start Consumer from command palette

**Comment by fbricon** [comment]
Thanks @angelozerr!

**Comment by angelozerr** [comment]
Thanks @fbricon for the review!

**Pull request merged**